mkFit/MkBuilder.cc

```diff
@@ -1279,55 +1279,65 @@ void MkBuilder::score_tracks(TrackVec& tracks)
 void MkBuilder::find_duplicates(TrackVec& tracks)
 {
   const auto ntracks = tracks.size();
+  float eta1, phi1, pt1, deta, dphi, maxpt, dr2;
   for (auto itrack = 0U; itrack < ntracks-1; itrack++)
   {
     auto & track = tracks[itrack];
-    float eta1 = track.momEta();
-    float phi1 = track.momPhi();
-    float pt1  = track.pT();
+    eta1 = track.momEta();
+    phi1 = track.momPhi();
+    pt1  = track.pT();
     for (auto jtrack = itrack+1; jtrack < ntracks; jtrack++)
     {
       auto & track2 = tracks[jtrack];
       if(track.label() == track2.label()) continue;
-      float eta2 = track2.momEta();
-      float phi2 = track2.momPhi();
-      float pt2  = track2.pT();
-      float dphi = squashPhiMinimal(phi1-phi2);
-      float deta = std::abs(eta2 - eta1);
-      float maxpt = std::max(pt1,pt2);
-      if(maxpt ==0) continue;
-      if(dphi < Config::maxdPhi && std::abs(pt2 - pt1)/maxpt < Config::maxdPt && deta < Config::maxdEta)
+     
+      deta = std::abs(track2.momEta() - eta1);
+      if(deta > Config::maxdEta) continue;
+
+      dphi = std::abs(squashPhiMinimal(phi1-track2.momPhi()));
+      if(dphi > Config::maxdPhi) continue;
+
+      dr2 = dphi*dphi + deta*deta;
+      if(dr2 < Config::maxdRSquared)
       {
-	if(Config::useHitsForDuplicates)
-	{
-	  std::vector<int> vecOfHits;
-	  float numHitsShared = 0;
-	  for (int ihit = 0; ihit < track.nTotalHits(); ihit++)
-	  {
-	    vecOfHits.push_back(track.getHitIdx(ihit));
-	  }
-	  for (int ihit2 = 0; ihit2 < track2.nTotalHits(); ihit2++)
+        //Keep track with best score
+        if(track.score() > track2.score())  track2.setDuplicateValue(true);
+	else track.setDuplicateValue(true);
+	continue;
```

Comment:
Indentation seems to be off here (possibly because of using tabs here while the surroundings uses spaces).